pkg/networkmanager/app.jsx

```diff
@@ -0,0 +1,135 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2021 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+import '../lib/patternfly/patternfly-cockpit.scss';
+import cockpit from "cockpit";
+import ReactDOM from "react-dom";
+import React from 'react';
+
+import { EmptyStatePanel } from "cockpit-components-empty-state.jsx";
+import { Button } from "@patternfly/react-core";
+import { ExclamationCircleIcon } from "@patternfly/react-icons";
+
+import { NetworkPage } from './network-main.jsx';
+import { NetworkInterfacePage } from './network-interface.jsx';
+import { init as initJQ, NetworkManagerModel } from './interfaces.js';
+import * as service from 'service.js';
+
+const _ = cockpit.gettext;
+
+class App extends React.Component {
```

Comment:
What about using a function here instead of a class, and use the hooks from lib/hooks.js?  See pkg/users/ for an extended example.

Reply:
Ack i will try it out

Reply:
I was not able to use the hooks here, I needed a callback for the useEvent, and also there were just many event listeners which updates the state, and would be hard to replace.
If you feel confortable feel free to jump in and try replacing the classes with functional components + use of the hooks.

Reply:
I had a go at this, what do you think?